File: src/01-easy/ascii-art/solution.fs
open System
let L = Console.ReadLine() |> int
let H = Console.ReadLine() |> int
let T = Console.ReadLine().ToUpper()
let rows = [| for _ in 0..H-1 -> Console.ReadLine() |]
for i in 0..H-1 do
    T |> Seq.iter (fun c ->
        let idx = int c - 65 in let idx = if idx<0||idx>25 then 26 else idx
        printf "%s" (rows.[i].Substring(idx*L, L)))
    printfn ""

File: src/01-easy/ascii-art/solution.kt
fun main(){
    val L=readLine()!!.trim().toInt();val H=readLine()!!.trim().toInt()
    val T=readLine()!!.uppercase()
    val rows=Array(H){readLine()!!}
    for(i in 0 until H){
        val sb=StringBuilder()
        for(c in T){var idx=c-'A';if(idx<0||idx>25)idx=26;sb.append(rows[i].substring(idx*L,idx*L+L))}
        println(sb)
    }
}
